internal/controllers/clusterrequest_controller.go

```diff
@@ -805,35 +832,115 @@ func (t *clusterRequestReconcilerTask) handleClusterPolicyConfiguration(ctx cont
 		}
 		targetPolicies = append(targetPolicies, *targetPolicy)
 	}
+	err = t.updateConfigurationAppliedStatus(
+		ctx, targetPolicies, allPoliciesCompliant, nonCompliantPolicyInEnforce)
+	if err != nil {
+		return false, err
+	}
+
+	// If there are policies that are not Compliant, we need to requeue and see if they
+	// time out or complete.
+	return nonCompliantPolicyInEnforce, nil
+}
 
-	// Update the ConfigurationApplied condition.
-	if allPoliciesCompliant {
+// hasPolicyConfigurationTimedOut determines if the policy configuration for the
+// ClusterRequest has timed out.
+func (t *clusterRequestReconcilerTask) hasPolicyConfigurationTimedOut(ctx context.Context) bool {
+	policyTimedOut := false
+	// Get the ConfigurationApplied condition.
+	configurationAppliedCondition := meta.FindStatusCondition(
+		t.object.Status.Conditions,
+		string(utils.CRconditionTypes.ConfigurationApplied))
+
+	// If the condition does not exist, set the non compliant timestamp since we
+	// get here just for policies that have a status different from Compliant.
+	if configurationAppliedCondition == nil {
+		t.object.Status.ClusterInstanceRef.NonCompliantAt = metav1.Now()
+		return policyTimedOut
+	}
+
+	// If the current status of the Condition is false.
+	if configurationAppliedCondition.Status == metav1.ConditionFalse {
+		switch configurationAppliedCondition.Reason {
+		case string(utils.CRconditionReasons.InProgress):
+			// Check if the configuration application has timed out.
+			if time.Since(t.object.Status.ClusterInstanceRef.NonCompliantAt.Time) > time.Duration(t.object.Spec.Timeout.Configuration) {
```

Comment:
Multiply time.Duration(t.object.Spec.Timeout.Configuration) by unit (*time.Second?) also indicate unit in definition of t.object.Spec.Timeout.Configuration object. It is currently in nanosconds.

Reply:
Yup, will add *time.Minute